File: Coding-Experiments/1_two_sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        ss= set()
        for j,i in enumerate(nums):
            if target-i in ss:
                return [nums.index(target-i),j]
            ss.add(i)
        return []

File: Coding-Experiments/1_two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int l = nums.size();
        std::unordered_set<int> mums;
        for (int i = 0; i < l; i++){
            if (mums.find(target-nums[i]) != mums.end()){
                auto f = std::ranges::find(nums,target-nums[i]);
                int j = std::distance(nums.begin(),f);
                return {i,j};
            }
            else{
                mums.insert(nums[i]);
            }
        }
        return {};
    }
};